File: programmers/Level 2/카펫.py
def solution1(brown, yellow):
    answer = []
    check = (brown - 4) // 2 # 꼭짓점 제외, 가로와 세로만 고려

    for i in range(check):
        if i * (check - i) == yellow:
            answer.append(check - i + 2)
            answer.append(i + 2)
            break

    return answer


def solution2(brown, yellow):
    answer = []
    sum = brown + yellow

    for i in range(3, sum // 2):
        if not sum % i:
            j = sum // i

            if yellow == (i - 2) * (j - 2):
                return [j, i]


def solution3(brown, yellow):
    answer = []

    for i in range(yellow):
        if not yellow % (i + 1):
            a, b = i + 1, yellow // (i + 1)
            if brown == 4 + (a * 2) + (b * 2):
                return [b + 2, a + 2]

File: programmers/Level 2/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int n = (brown - 4) / 2; // 꼭짓점 제외, 가로와 세로만 고려
    for(int i = 1; i < n; i++){
        int j = n - i;
        int m = i * j;
        if (m == yellow){
            answer.push_back(j + 2);
            answer.push_back(i + 2);
            break;
        }
    }
    return answer;
}
